topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -1309,8 +1323,12 @@ public void setKafkaTopics(List<KafkaTopic> ktList) {
             });
         }).compose(reconcileState -> {
             List<Future> futs = new ArrayList<>();
+            pausedTopicCounter.set(0);
```

Comment:
How can we be sure it won't get out of sync (e.g. with event-based reconciliation happening concurrently)?

Reply:
The count is refreshed with each periodic reconciliation. So there may be the "lag" when paused topic is created or the topic is paused. This lag should be at max the periodic reconciliation time. Do you think we should decrease/increase the counter everytime the topic is un/paused?